object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -450,14 +450,23 @@ public boolean isSubtype(final AnnotationMirror subAnno, final AnnotationMirror
       }
 
       if (AnnotationUtils.areSameByClass(subAnno, CalledMethodsPredicate.class)) {
+        String subPredicate =
+            AnnotationUtils.getElementValue(subAnno, "value", String.class, false);
         if (AnnotationUtils.areSameByClass(superAnno, CalledMethodsPredicate.class)) {
-          // Permit this only if the predicates are identical, to avoid complicated
-          // predicate equivalence calculation. Good enough in practice.
-          String predicate1 =
+          String superPredicate =
               AnnotationUtils.getElementValue(superAnno, "value", String.class, false);
-          String predicate2 =
-              AnnotationUtils.getElementValue(subAnno, "value", String.class, false);
-          return predicate1.equals(predicate2);
+          // shortcut if they're equal (most common case) to avoid calling the SMT solver
+          if (superPredicate.equals(subPredicate)) {
+            return true;
+          } else {
+            return CalledMethodsPredicateEvaluator.implies(subPredicate, superPredicate);
+          }
+        } else if (AnnotationUtils.areSameByClass(superAnno, CalledMethods.class)) {
+          // If the supertype is a called methods type, treat it as a conjunction of its
+          // arguments and use the predicate evaluator.
+          List<String> superVals = getValueOfAnnotationWithStringArgument(superAnno);
+          String superValPredicate = String.join(" && ", superVals);
+          return CalledMethodsPredicateEvaluator.implies(subPredicate, superValPredicate);
         }
         return false;
```

Comment:
I find it difficult to follow the logic of this method -- to verify that all the cases are handled.  Why is there a `return false;` here as a fallthrough to the possibilities of `superAnno` being `CalledMethodsPredicate` and `CalledMethods`, but there is none below at line 487 where the two cases are assumed to be exhaustive?  I would expect the two cases to be parallel.

Above, why do lines 442-446 have an else that returns false, but line 449 does not?  Again, I would expect the two cases to be parallel.